assets/js/components/Dashboard/NewPaMessage/StationsAndZones/SelectZonesPage.tsx

```diff
@@ -0,0 +1,541 @@
+import React, {
+  Fragment,
+  ReactNode,
+  useEffect,
+  useMemo,
+  useState,
+} from "react";
+import { Page } from "../types";
+import { Button } from "react-bootstrap";
+import { Place } from "Models/place";
+import fp from "lodash/fp";
+import { Screen } from "Models/screen";
+import { getZoneLabel, signIDs, sortByStationOrder } from "../../../../util";
+import cx from "classnames";
+import { ArrowLeftShort, ArrowRightShort, Dot } from "react-bootstrap-icons";
+import { useRouteToRouteIDsMap } from "../hooks";
+import { Dictionary } from "lodash";
+
+const ROUTE_TO_CLASS_NAMES_MAP: { [key: string]: string } = {
+  Green: "bg-green",
+  Red: "bg-red",
+  Orange: "bg-orange",
+  Blue: "bg-blue",
+  Mattapan: "bg-mattapan",
+  Silver: "bg-silver",
+  Bus: "bg-bus",
+};
+
+interface Props {
+  value: string[];
+  onChange: (signIds: string[]) => void;
+  onSubmit: (signIds: string[]) => void;
+  navigateTo: (page: Page) => void;
+  places: Place[];
+}
+
+const LEFT_ZONES = ["s", "w"] as const;
+const MIDDLE_ZONES = ["c", "m"] as const;
+const RIGHT_ZONES = ["n", "e"] as const;
+
+const SelectZonesPage = ({
+  value,
+  onChange,
+  onSubmit,
+  navigateTo,
+  places,
+}: Props) => {
+  const [selectedRouteFilter, setSelectedRouteFilter] = useState("");
+  const [newPlaces, setNewPlaces] = useState<Place[]>([]);
+  const [selectedRoutes, setSelectedRoutes] = useState<Dictionary<string[]>>(
+    {},
+  );
+  const routeToRouteIDMap = useRouteToRouteIDsMap();
+
+  const isSelected = (id: string) => value.includes(id);
+
+  const directionLabels = useMemo(() => {
+    switch (selectedRouteFilter) {
+      case "Green":
+      case "Green-B":
+      case "Green-C":
+      case "Green-D":
+      case "Green-E":
+      case "Blue":
+        return {
+          left: "Westbound",
+          middle: "Westbound & Eastbound",
+          right: "Eastbound",
+        };
+      case "Silver":
+        return { left: "To South Station/Downtown", right: "Eastbound" };
+      case "Bus":
+        return { left: "", middle: "", right: "" };
+      default:
+        return {
+          left: "Southbound",
+          middle: "Southbound & Northbound",
+          right: "Northbound",
+        };
+    }
+  }, [selectedRouteFilter]);
+
+  const massSelectButtonLabels = useMemo(() => {
+    if (selectedRouteFilter === "Silver") {
+      return {
+        left: "Both",
+        middle: "South Station/Downtown",
+        right: "Outbound Content",
+      };
+    }
+
+    return {
+      left: "All Zones (except bus)",
```

Comment:
**question:** is this still accurate if you select only bus from the select stations page?

Reply:
These buttons are not visible when the selected route filter is `Bus`.